libs/librrgraph/src/base/rr_graph_storage.h

```diff
@@ -545,11 +545,18 @@ class t_rr_graph_storage {
     // Reserve at least num_edges in the edge backing arrays.
     void reserve_edges(size_t num_edges);
 
-    // Add one edge.  This method is efficient if reserve_edges was called with
-    // the number of edges present in the graph.  This method is still
-    // amortized O(1), like std::vector::emplace_back, but both runtime and
-    // peak memory usage will be higher if reallocation is required.
-    void emplace_back_edge(RRNodeId src, RRNodeId dest, short edge_switch);
+    /***
+     * @brief Add one edge.  This method is efficient if reserve_edges was called with
+     * the number of edges present in the graph.  This method is still
+     * amortized O(1), like std::vector::emplace_back, but both runtime and
+     * peak memory usage will be higher if reallocation is required.
+     * @ param remapped This is used later in remap_rr_node_switch_indices to check whether an
+     * end needs to be remapped. This remapping is done to change the switch index from arch_sw_idx
```

Comment:
end needs to be remapped -> edge needs its switch ID remapped from the arch_sw_idx to rr_sw_idx.
flight-weighted -> fly-weighted

Add: "We need to control this remapping, as it must be done when we build an rr-graph from an arch file, but should not be done when we read in an entire rr-graph from a file (as the file already used rr_sw_idx values).